lib/services/profiles-api-service.js

```diff
@@ -81,6 +84,14 @@ function profileApiServiceFactory(
                         query.mac
                     );
                 }
+            }).then(function () {
+                if ( req.get(Constants.HttpHeaders.ApiProxyIp) ) {
+                    var proxy =  'http://%s:%s'.format(
+                        req.get(Constants.HttpHeaders.ApiProxyIp),
+                        req.get(Constants.HttpHeaders.ApiProxyPort)
+                    );
+                    return waterline.lookups.upsertProxyToMacAddress(proxy, query.mac);
+                }
```

Comment:
@brianparry Yeah I see what you mean.  So should setLookup return the node only if it added it to the database and return undefined if it did not add anything?  It looks like that is what the code was doing previously.